lib/bookmarks.js

```diff
@@ -0,0 +1,39 @@
+import bookmarkModel from '../models';
+
+const { bookmarks, articles, users } = bookmarkModel;
+
+/** @description validate bookmarks
+   * @param {integer} userId is the request parameter
+   * @param {integer} articleId is the response parameter
+   * @return {object} the response object
+   * @public
+   */
+const validateBookmarks = async (userId, articleId) => {
+  try {
+    const user = await users.findOne({
+      where: {
+        id: userId
+      }
+    });
+    const article = await articles.findOne({
+      where: {
+        id: articleId
+      }
+    });
+    const isAvalaible = await bookmarks.findOne({
+      where: {
+        userId,
+        articleId
+      }
+    });
+
+    if (!article || isAvalaible || !user) {
```

Comment:
Please explain this condition as it applies to your work `!article || isAvalaible || !user`

Reply:
- !article: am checking if the article that is trying to be bookmarked exist, maybe it may have been deleted already, so am checking if the article exists in the database first.
- isAvalaible: am checking if the article has already been bookmarked by the user that is trying to bookmark it, to avoid having the same bookmark copy in our database
-!user: am checking if the user that is creating a new bookmark exist in the database, cause I found out that the token might be valid but the user does not exist.
@Chidi3b3r3